src/Nox.Types/Types/DateTime/DateTime.cs

```diff
@@ -0,0 +1,275 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Nox.Types;
+public sealed class DateTime : ValueObject<System.DateTime, DateTime>
+{
+    private DateTimeTypeOptions _dateTimeTypeOptions = new();
+
+    public int Year
+    {
+        get { return Value.Year; }
+        private set { Value = new System.DateTime(value, Value.Month, Value.Day, Value.Hour, Value.Minute, Value.Second); }
+    }
+    public int Month
+    {
+        get { return Value.Month; }
+        private set => Value = new System.DateTime(Value.Year, value, Value.Day, Value.Hour, Value.Minute, Value.Second);
+    }
+    public int Day
+    {
+        get { return Value.Day; }
+        private set { Value = new System.DateTime(Value.Year, Value.Month, value, Value.Hour, Value.Minute, Value.Second); }
+    }
+    public int Hour
+    {
+        get { return Value.Hour; }
+        private set { Value = new System.DateTime(Value.Year, Value.Month, Value.Day, value, Value.Minute, Value.Second); }
+    }
+    public int Minute
+    {
+        get { return Value.Minute; }
+        private set { Value = new System.DateTime(Value.Year, Value.Month, Value.Day, Value.Hour, value, Value.Second); }
+    }
+    public int Second
+    {
+        get { return Value.Second; }
+        private set { Value = new System.DateTime(Value.Year, Value.Month, Value.Day, Value.Hour, Value.Minute, value); }
+    }
+
+    public DateTime() { Value = System.DateTime.MinValue; }
+
+    /// <summary>
+    /// Creates and validates a new instance of <see cref="DateTime"/>./>.
+    /// </summary>
+    /// <param name="year"></param>
+    /// <param name="month"></param>
+    /// <param name="day"></param>
+    /// <param name="hour"></param>
+    /// <param name="minute"></param>
+    /// <param name="second"></param>
+    /// <returns>New <see cref="DateTime"/> object with default <see cref="DateTimeTypeOptions"/> .</returns> 
+    /// <exception cref="TypeValidationException"></exception>
+    public static DateTime From(int year, int month, int day, int hour = 0, int minute = 0, int second = 0) => From(new DateTimeTypeOptions(), year, month, day, hour, minute, second);
```

Comment:
There is no need to have these overloads, as you can use the
`From(System.DateTime dateTime)`
and just instantiate System.DateTime based on year, month, and day.

Reply:
Overload with the dates as integers removed.